pl_examples/domain_templates/computer_vision_fine_tuning.py

```diff
@@ -0,0 +1,440 @@
+        return {'log': {'val_loss': val_loss_mean,
+                        'val_acc': val_acc_mean,
+                        'step': self.current_epoch}}
+
+    def configure_optimizers(self):
+        optimizer = optim.Adam(filter(lambda p: p.requires_grad,
+                                      self.parameters()),
+                               lr=self.hparams.lr)
+
+        scheduler = MultiStepLR(optimizer,
+                                milestones=self.hparams.milestones,
+                                gamma=self.hparams.lr_scheduler_gamma)
+
+        return [optimizer], [scheduler]
+
+    def prepare_data(self):
+        """Download images and prepare images datasets."""
+
+        # 1. Download the images
+        download_and_extract_archive(url=DATA_URL,
+                                     download_root=self.dl_path,
+                                     remove_finished=True)
+
+        data_path = Path(self.dl_path).joinpath('cats_and_dogs_filtered')
+
+        # 2. Load the data + preprocessing & data augmentation
+        normalize = transforms.Normalize(mean=[0.485, 0.456, 0.406],
+                                         std=[0.229, 0.224, 0.225])
+
+        train_dataset = ImageFolder(root=data_path.joinpath('train'),
+                                    transform=transforms.Compose([
+                                        transforms.Resize((224, 224)),
+                                        transforms.RandomHorizontalFlip(),
+                                        transforms.ToTensor(),
+                                        normalize,
+                                    ]))
+
+        valid_dataset = ImageFolder(root=data_path.joinpath('validation'),
+                                    transform=transforms.Compose([
+                                        transforms.Resize((224, 224)),
+                                        transforms.ToTensor(),
+                                        normalize,
+                                    ]))
+
+        self.train_dataset = train_dataset
+        self.valid_dataset = valid_dataset
+
+    def __dataloader(self, train):
+        """Train/validation loaders."""
+
+        _dataset = self.train_dataset if train else self.valid_dataset
+        loader = DataLoader(dataset=_dataset,
+                            batch_size=self.hparams.batch_size,
+                            num_workers=self.hparams.num_workers,
+                            shuffle=True if train else False)
+
+        return loader
+
+    def train_dataloader(self):
+        log.info('Training data loaded.')
+        return self.__dataloader(train=True)
+
+    def val_dataloader(self):
+        log.info('Validation data loaded.')
+        return self.__dataloader(train=False)
+
+    @staticmethod
+    def add_model_specific_args(parent_parser):
+        parser = argparse.ArgumentParser(parents=[parent_parser])
+        parser.add_argument('--backbone',
```

Comment:
use `add_argparse_args` so we limit duplication and add just the new/needed for a model

Reply:
by "limit code duplication", you want me to remove [this line](https://github.com/PyTorchLightning/pytorch-lightning/pull/1564/files/ff19e3763919c38595731086ed8cd5a197b12ee8#diff-78fcc134908f8b115ba9b07e9fc99732R427)?

Reply:
I mean remove lines which are generated from Trainer arguments... does it make sense?

Reply:
https://github.com/PyTorchLightning/pytorch-lightning/blob/9b86aea98bdaa73bc3bf8841f4dc794f46a3f2ac/pytorch_lightning/trainer/trainer.py#L589